R/backend-access.R

```diff
@@ -167,5 +167,22 @@ sql_escape_logical.ACCESS <- function(con, x) {
   y[is.na(x)] <- "NULL"
   y
 }
+                   
+#' @export                   
+sql_escape_date.ACCESS <-  function(con, x) {
+  # Access delimits dates using octothorpes, and uses YYYY-MM-DD
+  y <- format(x, "#%F#")
+  y[is.na(x)] <- "NULL"
+  y
+}
+
+#' @export
+sql_escape_datetime.ACCESS <-  function(con, x) {
+  # Access delimits datetimes using octothorpes, and uses YYYY-MM-DD HH:MM:SS
+  # Timezones are not supported in Access
+  y <- format(x, "#%F %T#")
```

Comment:
What does not supporting time zones imply here? Should we always generate a time in the system time zone?

Reply:
It means timezone information is ignored when you escape a value that includes timezone information, e.g. `escape(as.POSIXct('2020-01-01 01:01:01', tz = "US/Alaska"), con = simulate_access())` will result in the same thing as `escape(as.POSIXct('2020-01-01 01:01:01', tz = "Europe/Berlin"), con = simulate_access())`. 

Some RDBMSes have a concept of timezones (e.g. SQL Server, Oracle) and when escaping a datetime field the timezone could be included, however, Access does not. 

I assume that's a non-issue, especially since escapes for those databases (e.g.  `escape(as.POSIXct('2020-01-01 01:01:01', tz = 'US/Alaska'), con = simulate_mssql())`) always provides the zulu timezone and also ignore timezone information.  But that's a separate issue, and one that's considerably harder to address and might break existing code when addressed.

Reply:
Ah, got it. Thanks for the explanation!